src/main/java/seedu/address/model/person/Appointment.java

```diff
@@ -301,6 +301,50 @@ public String getRecurringStatus() {
         return recurring;
     }
 
+    /**
+     * Returns the group number where this appointment belongs to, which is determined
+     * by its tags.
+     *
+     */
+    public Integer getGroupNumber() {
+        if (this.getTags().isEmpty()) {
+            return 0;
+        }
+        int value = 0;
+        for (Tag tag: this.getTags()) {
+            value += (tag.ordinal() == 0 ? 1 : tag.ordinal() == 1 ? 2 : 4);
+        }
+        return value;
+    }
+
+    /**
+     * Returns -1 if this appointment appears before the other appointment, and
+     * returns 0 if this appointment has the same order as the other appointment, and
+     * returns 1 if this appointment appears after the other appointment.
+     *
+     * @param appointment The other appointment to compare with.
+     */
+    public int compareTo(Appointment appointment) {
+        return this.dateTime.isBefore(appointment.dateTime)
+                ? -1
+                : this.dateTime.isAfter(appointment.dateTime)
+                ? 1
+                : 0;
+    }
+
+    /**
+     * Returns 10 * group difference -1 if this appointment appears before the other appointment, and
+     * returns 10 * group difference if this appointment has the same order as the other appointment, and
+     * returns 10 * group difference + 1 if this appointment appears after the other appointment. This method
+     * makes sure that appointments with the same tag group are grouped together.
+     *
+     * @param appointment The other person to compare with.
+     */
+    public int groupCompareTo(Appointment appointment) {
+        return 10 * this.getGroupNumber().compareTo(appointment.getGroupNumber())
+                + this.compareTo(appointment);
+    }
```

Comment:
```suggestion
    public int groupCompareTo(Appointment appointment) {
        int tagWeight = 10;
        int dateWeight = 1;
        return tagWeight * this.getGroupNumber().compareTo(appointment.getGroupNumber())
                + dateWeight * this.compareTo(appointment);
    }
```
If I'm understanding this method correctly, I believe that the "10" is there to act as a weight to give tags higher priority than the dates? In that case, perhaps it may be better to add a constant to reduce the presence of magic literals and make it more intuitive?

Reply:
Yea your understanding is correct, and sure it is better to add constants to make it more intuitive! Thank you for pointing it out!